File: code-samples/clean-architecture-java/src/main/java/com/example/adapter/DatabaseGateway.java
package com.example.adapter;

import com.example.usecase.ArticleGateway;
import com.example.entity.Article;
import com.example.entity.ArticleId;

public class DatabaseGateway implements ArticleGateway {

    @Override
    public Article getArticle(ArticleId articleId) {
        // Connect database
        return new Article("Compfest - Tech to Elevate");
    }
}

File: code-samples/clean-architecture-kotlin/src/main/kotlin/com/example/adapter/DatabaseGateway.kt
package com.example.adapter

import com.example.usecase.ArticleGateway
import com.example.entity.Article
import com.example.entity.ArticleId

class DatabaseGateway : ArticleGateway {
    override fun getArticle(articleId: ArticleId): Article {
        return Article("Compfest - Tech to Elevate")
    }
}
